### DISCUSSION TITLE
How to send two separated data to two different collaborators?

### DISCUSSION THREAD

**Discussion by amin-kaveh** [securefederatedai/openfederatedlearning#778] [Q&A]
Hi, 

I am new in federated learning and openfl. I have two separated (let's call them private) datasets and I want to send them to each collaborators separately. When I review tutorials, it seems that `FederatedDataSet` gets all the data together and split them into multiple sets and then `FederatedModel` function sends the same model and splits of data to each collaborators. 

This is the code that have have from tutorials:
```
fl_data = FederatedDataSet(train_data, train_labels, valid_data, valid_labels, batch_size, num_classes)
fl_model = FederatedModel(build_model, optimizer, loss_fn, data_loader= fl_data)

```
But suppose I have private data that I want to sent to each collaborators:
data for collaborator 1: `train_data_1, train_label_1, valid_data_1, valid_label_1`
data for collaborator 2: `train_data_2, train_label_2, valid_data_2, valid_label_2`

Now, I want to send the model and data to each collaborators separately and then train them. Something like this:
```
send(build_model, train_data_1, train_label_1, valid_data_1, valid_label_1, to_collaborator_1 )
send(build_model, train_data_2, train_label_2, valid_data_2, valid_label_2, to_collaborator_2 )

```
And then start training them.
What are those commands and how can I train the model?
 
I really appreciate any help and insight.

**Comment by mansishr** [comment]
Hi @amin-kaveh, welcome to OpenFL! In general, federated learning assumes that the collaborators are data owners so they already have a copy of the local/private dataset with them. Do you mean that in your case, the collaborators cannot access data?

**Comment by AtyDany** [comment]
hi amin
did you find any answer for your question, i have same problem, however chatgpt suggested bellow code which is not working:

 # Initialize an OpenFL workspace
workspace = fx.workspace()
workspace.add_model(hospital1_model, 'hospital1')
workspace.add_model(hospital2_model, 'hospital2')
workspace.add_model(hospital3_model, 'hospital3')

# Run the federated learning plan
workspace.run_plan(federated_learning_plan)

# Evaluate the global model on the test set
global_model = workspace.get_model()
test_loss, test_accuracy = global_model.evaluate(X_test,  y_test)
print(f'Global Model Test Accuracy: {test_accuracy}')
